README.md

```diff
@@ -74,16 +74,54 @@ It is possible for Cannon to verify freshly deployed contracts, using the follow
+
+1. Build the cannonfile in the contracts repo as normal.
+
+2. In the deployments repo (this repo), update the deployment cannonfile to use the new version of the locally built cannonfile. This means setting the `source` and `target` of the clone step to use the package name of the cannonfile from the contracts repo. e.g. `source = "infinex-governance-points:2.0.0@stw"`
+
+3. When building the new deployment, cannon will think that it needs to retry some of the steps which it has already completed in the previous deployment. To solve this, we use `cannon alter` to tell the new deployment that it doesn't need to do those steps. This is the command you'll need to run:
+
+```
+pnpm cannon alter --subpkg [CLONE_STEP_FROM_DEPLOYMENT_FILE] [PACKAGE_TO_UPGRADE_FROM] mark-complete [CANNON_STEP_TO_SKIP] --chain-id [CHAIN_ID_OF_DEPLOYMENT]
+```
+
+Let's break down what this command is doing. We are running `cannon alter` on the `PACKAGE_TO_UPGRADE_FROM`. What this does is marks the steps from the `PACKAGE_TO_UPGRADE_FROM` that don't need to be repeated when upgrading from it. You need to specify the `CLONE_STEP_FROM_DEPLOYMENT_FILE` to tell cannon where to find the `CANNON_STEP_TO_SKIP`, since this is where the step is being imported from.
+
+
+e.g. To mark the `GovernancePointsProxy` deploy step in the previous deployment so that it doesn't attempt it again in the new deployment, run this command:
+
+```
+pnpm cannon alter --subpkg clone.InfinexGovernancePoints infinex-gp:0.1.7@StW mark-complete deploy.GovernancePointsProxy --chain-id 84532
+```
+
+At no point should you be referencing the locally built cannonfile. This should only be referenced in the deployment file in the source/target of the clone.
+
+4. The output of the alter command should be an ipfs url. If it comes up with `Cannot read properties of null`, that means you've got an invalid reference somewhere. Now you are ready to run the `build` command using the output from the `alter` command.
+
+```
+pnpm cannon build [FILE_PATH_OF_DEPLOYMENT_FILE] --chain-id [CHAIN_ID_OF_DEPLOYMENT] --registry-priority local --upgrade-from @ipfs:[IPFS_CONTENT_HASH] --private-key [PRIVATE_KEY] 
+```
+
+Let's break down what this command is doing. `cannon build` is deploying the `FILE_PATH_OF_DEPLOYMENT_FILE` cannonfile to the `CHAIN_ID_OF_DEPLOYMENT` specified. When it does so, we want the steps it's importing from the `clone` source/target to be sourced from our local `registry-priority` so it finds our newly built local package, rather than try to look for it on the cannon registry, because it hasn't been published yet. We specify `upgrade-from` to tell cannon that we will be upgrading this from an existing deployment and that it doesn't need to run all the steps. It looks at `IPFS_CONTENT_HASH`, the output of `alter`, to know which steps it doesn't need to repeat. `PRIVATE_KEY` is the key used to deploy these contracts.
```

Comment:
In this scenario, even if the package is published, the registry priority local is still required for the generated ipfs hash, without it I was told it would fail.

Reply:
please remove the private key references, as we should be using frame going forward